### DISCUSSION TITLE
Взаимодействие firefox и tpws

### DISCUSSION THREAD

**Discussion by dodecube** [bol-van/zapret#915] [Q&A] [answered]
Есть пара вопросов, а конкретно:
Как работает обычный tpws, тоесть я понимаю что это transparent proxy, но какой именно http, https?
Как заставить вместе работать firefox и tpws, мне требуется что-бы в firefox работал ютуб но я не знаю что лучше использовать для подключения tpws прокси к firefox'у.

**Comment by bol-van** [comment, marked as answer]
tpws может работать и в transparent, и в socks режиме.
В transparent не является ни http, ни https proxy. Он является tcp proxy и требует метаинформации от ядра ОС для получения оригинального ip адреса назначения. Перенаправить с другой системы через DNAT не выйдет как на squid.

В фоксе можно начать с socks режима, тк он проще.
Но лучше все же найти стратегию для GGC серверов сначала и опробовать ее через curl

**Reply by dodecube** [reply]
спасибо, уже получилось настроить через socks, это круто что автор такой программы общается с пользователями
